### PR TITLE
feat: tensorflow -> PyTorch로 수정, 전체 클래스(345개) 모델 생성

### PR DESCRIPTION
**Pull request by brothergiven**
## PyTorch를 선택한 이유

1. OCR(텍스트 인식) 을 위한 CRAFT 패키지는 torchvision 0.12.0 이하에서만 작동한다.
2. 꽤 옛날버전인 torchvision 0.12.0 버전을 맞추기 위해 tensorflow 버전을 계속 낮춰보았으나, 패키지 버전이 계속 충돌하는 문제 발생(특히 numpy)
3. PyTorch와 Tensorflow를 함께 사용해도 GPU는 하나의 모듈에만 사용될 수 있음. 그리고 PyTorch와 Tensorflow를 같은 환경에서 사용하는 경우는 아무리 구글링을 해도 찾을 수 없었다. 

### 1. 학습 결과

pr_img_1

- Epoch(학습 반복 횟수)당 Accuracy(정확도) 변화
- 10번 학습, Accuracy 약 60%
- Epoch와 데이터셋을 늘려가며 Accuracy 변화를 계속 지켜봐야겠다(현재는 클래스당 최대 1000개 데이터 사용)

pr_img_2
    
- Epoch당 Loss(정답이랑 차이) 변화
- 1.7정도 까지 내려감, 최소 0.x 까지는 내려야 함

### 2. Test - 한글 텍스트 함께

pr_img_3

pr_img_4

pr_img_5

pr_img_6

- 진짜 hurricane을 넣어보자

### Test 2 - Hurricane

pr_img_7

pr_img_8

pr_img_9

### Test 3 - 영어 텍스트 포함

pr_img_10

pr_img_11

pr_img_12

pr_img_13

## 발전방향

1. 모델 Accuracy를 적어도 90까지는 올려야 함
2. CRAFT 텍스트 인식도 믿을만한 수준은 아님(꽤 정확한 편이지만, 가끔 낙서를 이미지로 인식하고 마스킹해버리는 경우가 있음)
3. 배포 방법을 잘 생각해야 한다. 연산속도가 오래 걸리고(특히 이미지 추출) CPU 자원도 많이 잡아먹을 것으로 예상됨.. 계속 알아봐야겠다

### CODE REVIEW

**Comment by brothergiven** [comment]
https://beeny-ds.tistory.com/entry/%EC%86%8C%EA%B0%9C-ONNX-%EB%9E%80
